src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1191,13 +1198,14 @@ bool PhaseIterGVN::verify_Value_for(Node* n, bool strict) {
   tnew->dump_on(&ss);
   ss.cr();
   tty->print_cr("%s", ss.as_string());
-  return true;
+
+  assert(false, "Missed Value optimization opportunity in PhaseIterGVN for %s", n->Name());
```

Comment:
What if it gets called during CCP? Then it is not just a missed opportunity, but possibly a correctness problem.

I wonder if we should have different assert messages here. We could even just pass a string into the method, either `IGVN` or `CCP`.

What do you think?